eth/gaspricemonitor.go

```diff
@@ -45,6 +45,9 @@ func NewGasPriceMonitor(gpo GasPriceOracle, pollingInterval time.Duration, minGa
 	minGasP := big.NewInt(0)
 	if minGasPrice != nil {
 		minGasP = minGasPrice
+		if monitor.Enabled {
+			monitor.MinGasPrice(minGasPrice)
```

Comment:
I think this should be called unconditionally regardless of the value of the `minGasPrice` argument passed in the constructor. This way if the `minGasPrice` argument is nil we would still record the current minGasPrice as 0 which is the default value.